statsmodels/stats/rates.py

```diff
@@ -0,0 +1,200 @@
+    Biometrical Journal 50 (2008) 2, 2008
+
+    '''
+
+    # shortcut names
+    y1, n1, y2, n2 = count1, exposure1, count2, exposure2
+    d = n2 / n1
+    r = ratio_null
+    r_d = r / d
+
+    if method in ['score']:
+        stat = (y1 - y2 * r_d) / np.sqrt((y1 + y2) * r_d)
+        dist = 'normal'
+    elif method in ['wald']:
+        stat = (y1 - y2 * r_d) / np.sqrt(y1 + y2 * r_d**2)
+        dist = 'normal'
+    elif method in ['sqrt']:
+        stat = 2 * (np.sqrt(y1 + 3 / 8.) - np.sqrt((y2 + 3 / 8.) * r_d))
+        stat /= np.sqrt(1 + r_d)
+        dist = 'normal'
+    elif method in ['exact-cond', 'cond-midp']:
+        from statsmodels.stats import proportion
+        bp = r_d / (1 + r_d)
+        y_total = y1 + y2
+        stat = None
+        # TODO: why y2 in here and not y1, check definition of H1 "larger"
+        pvalue = proportion.binom_test(y1, y_total, prop=bp, alternative=alternative)
+        if method in ['cond-midp']:
+            # not inplace in case we still want binom pvalue
+            pvalue = pvalue - 0.5 * stats.binom.pmf(y1, y_total, bp)
+
+        dist = 'binomial'
+
+    if dist == 'normal':
+        return _zstat_generic2(stat, 1, alternative)
+    else:
+        return stat, pvalue
+
+
+def etest_twopoisson(count1, exposure1, count2, exposure2, ratio_null=1,
+                         method='score', alternative='2-sided'):
+    """initial version of E-test for two sample Poisson comparison
+
+    """
+    y1, n1, y2, n2 = count1, exposure1, count2, exposure2
+    d = n2 / n1
+    r = ratio_null
+    r_d = r / d
+
+    if method in ['score']:
+        stat_func = lambda x1, x2:  (x1 - x2 * r_d) / np.sqrt((x1 + x2) * r_d)
+        rate2_cmle = (y1 + y2) / n2 / (1 + r_d)
+        rate1_cmle = rate2_cmle * r
+        rate1 = rate1_cmle
+        rate2 = rate2_cmle
+    elif method in ['wald']:
+        #stat_func = lambda x1, x2:  (x1 - x2 * r_d) / np.sqrt(x1 + x2 * r_d**2)
+        def stat_func(x1, x2):
+            return (x1 - x2 * r_d) / np.sqrt(x1 + x2 * r_d**2)
+        rate2_mle = y2 / n2
+        rate1_mle = y1 / n1
+        rate1 = rate1_mle
+        rate2 = rate2_mle
+
+    # The sampling distribution needs to be based on the null hypotheis
+    # use constrained MLE from 'score' calculation
+    rate2_cmle = (y1 + y2) / n2 / (1 + r_d)
+    rate1_cmle = rate2_cmle * r
+    rate1 = rate1_cmle
+    rate2 = rate2_cmle
+    mean1 = n1 * rate1
+    mean2 = n2 * rate2
+
+    stat_sample = stat_func(y1, y2)
+
+    # The following uses a fixed truncation for evaluating the probabilities
+    # It will currently only work for small counts, so that sf at truncation point
+    # is small
+    # We can make it depend on the amount of truncated sf.
+    # Some numerical optimization or checks for large means need to be added.
+    threshold = 100
+    threshold = stats.poisson.isf(1e-13, max(mean1, mean2))
+    threshold = max(threshold, 100)   # keep at least 100
+    y_grid = np.arange(threshold + 1)
```

Comment:
needs options, kwds, for users to change this, see also #6676
I'm not sure it's the same grid, i.e. here it says y_grid